### PR TITLE
Registry integration test: Test tags

### LINKED ISSUE #896: Registry integration tests: Tags
## Description

After the development of wazuh/wazuh#5576, new integration tests are going to be needed.
Each file may perform several tests. The structure of the directory will be:

### test_tags

| Test      | Action                             | Expected result                                                           |
| --------- | ---------------------------------- | ------------------------------------------------------------------------- |
| test_tags | Apply tags to monitored registries | Ensure the events raised for the monitored registry has the expected tags |

### PR DESCRIPTION
**Pull request by antoniomanuelfr**
Hello team,

This PR aims to add a new integration test in order to check that the `tag`  option is working correctly.

Closes #896

# Tests logic
- The test will configure some keys using the tag option and it will create values in both keys.
- When the events are generated, it checks that the events have the `tag` field and its content is the same as the configured.
# Tests checks

- [X] Proven that tests **pass** when they have to pass
- [X] Proven that tests **fail** when they have to fail

## Windows

- When the expected behavior occurs:

```
python -m pytest -vvx test_tags\
platform win32 -- Python 3.7.3, pytest-6.0.2, py-1.9.0, pluggy-0.13.0 
==============================================================
[  2%] test_registry_tags.py PASSED
[  6%] test_registry_tags.py PASSED
[ 13%] test_registry_tags.py PASSED
[ 20%] test_registry_tags.py PASSED
[ 26%] test_registry_tags.py PASSED
[ 33%] test_registry_tags.py PASSED
[ 40%] test_registry_tags.py PASSED
[ 46%] test_registry_tags.py PASSED
[ 53%] test_registry_tags.py PASSED
[ 60%] test_registry_tags.py PASSED
[ 66%] test_registry_tags.py PASSED
[ 73%] test_registry_tags.py PASSED
[ 80%] test_registry_tags.py PASSED
[ 86%] test_registry_tags.py PASSED
[ 93%] test_registry_tags.py PASSED
[100%]
======================= 15 passed in 62.16s (0:01:02) ==================
```
Best regards.

### CODE REVIEW

File: tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py [hunk 1]
```diff
@@ -48,22 +41,35 @@ def get_configuration(request):
...
+        Name of the file to be created.
+    content : str, bytes
+        Content to fill the new file.
+    """
+    defined_tags = get_configuration['metadata']['fim_tags']
+
+    def tag_validator(event):
+        assert defined_tags == event['data']['tags'], f'defined_tags are not equal'
+
+
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py:69]: There are too many blank lines before `registry_value_cud`, remove one of them.
```suggestion
```

File: tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py [hunk 2]
```diff
@@ -48,22 +41,35 @@ def get_configuration(request):
...
+    registry_value_cud(key, subkey, wazuh_log_monitor, arch=arch,
+                     time_travel=get_configuration['metadata']['fim_mode'] == 'scheduled',
+                     min_timeout=global_parameters.default_timeout,
+                     triggers_event=True,
+                     validators_after_cud=[tag_validator]
+                    )
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py:75]: These lines are under-indented.
```suggestion
                       time_travel=get_configuration['metadata']['fim_mode'] == 'scheduled',
                       min_timeout=global_parameters.default_timeout,
                       triggers_event=True,
                       validators_after_cud=[tag_validator]
                       )
```

File: tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py [hunk 3]
```diff
@@ -6,38 +6,31 @@
 import pytest
 
 from wazuh_testing import global_parameters
-from wazuh_testing.fim import LOG_FILE_PATH, regular_file_cud, generate_params, delete_registry, registry_parser
+from wazuh_testing.fim import LOG_FILE_PATH, registry_value_cud, generate_params, delete_registry, registry_parser
 from wazuh_testing.tools import PREFIX
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py:10]: `PREFIX` is imported but unused.
```suggestion
```

File: tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py [hunk 4]
```diff
@@ -6,38 +6,31 @@
 import pytest
 
 from wazuh_testing import global_parameters
-from wazuh_testing.fim import LOG_FILE_PATH, regular_file_cud, generate_params, delete_registry, registry_parser
+from wazuh_testing.fim import LOG_FILE_PATH, registry_value_cud, generate_params, delete_registry, registry_parser
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py:9]: `delete_registry` and `registry_parser` are imported but unused.
```suggestion
from wazuh_testing.fim import LOG_FILE_PATH, registry_value_cud, generate_params
```

File: tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py [hunk 5]
```diff
@@ -6,38 +6,31 @@
...
 test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
-registr_str, registry2 = test_regs
-
-registry_list=[(sub_key_1, win32con.KEY_WOW64_64KEY), (sub_key_2, win32con.KEY_WOW64_32KEY),
-               (sub_key_2, win32con.KEY_WOW64_64KEY)]
-
-monitoring_modes = ['scheduled', 'scheduled']
-
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+test_regs = [ os.path.join(key, sub_key), os.path.join(key, sub_key_2)]
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py:24]: Remove whitespace after `[`.
```suggestion
test_regs = [os.path.join(key, sub_key), os.path.join(key, sub_key_2)]
```

File: tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py [hunk 6]
```diff
@@ -6,38 +6,31 @@
...
+reg1, reg2 = test_regs
 # Configurations
+tags = ['tag1', 'tág', '0tag', '000', 'a' * 1000]
+conf_params = {'WINDOWS_REGISTRY_1':reg1, 'WINDOWS_REGISTRY_2':reg2}
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py:28]: Missing whitespace after `:`.
```suggestion
conf_params = {'WINDOWS_REGISTRY_1': reg1, 'WINDOWS_REGISTRY_2': reg2}
```

File: tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py [hunk 7]
```diff
@@ -6,38 +6,31 @@
...
-conf_params = {'WINDOWS_REGISTRY_1': registr_str, 'WINDOWS_REGISTRY_2' : registry2}
-configurations_path = os.path.join(test_data_path, 'wazuh_conf_registry_both.yaml')
-p, m = generate_params(extra_params=conf_params, modes=monitoring_modes)
+configurations_path = os.path.join(test_data_path, 'wazuh_registry_tag_conf.yaml')
+p, m = generate_params(extra_params=conf_params,
+                       apply_to_all=({'FIM_TAGS': tag} for tag in tags),
+                       modes=['scheduled'])
 configurations = load_wazuh_configurations(configurations_path, __name__, params=p, metadata=m)
 
 # Fixtures
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_tags/test_registry_tags.py:36]: There should be two blank lines before a function declaration.
```suggestion
# Fixtures

```

**Review by jesusjimsa — CHANGES_REQUESTED** [review]
I have requested some style changes. I will run the tests tomorrow to confirm they work correctly and send a new revision if there is something wrong.

**Review by jesusjimsa — APPROVED** [review]
LGTM!

**Review by bah07 — APPROVED** [review]
LGTM!

**Pull request merged**